aiographite/aiographitesend.py

```diff
@@ -0,0 +1,163 @@
+from .protocol import PlaintextProtocol, PickleProtocol
+from .graphite_encoder import GraphiteEncoder
+import asyncio
+import pickle
+import struct
+import socket
+import os
+import time
+from typing import Dict, Tuple, List, Callable
+
+
+DEFAULT_GRAPHITE_PLAINTEXT_PORT = 2003
+DEFAULT_GRAPHITE_PICKLE_PORT = 2004
+SUPPORT_PROTOCOLS = ["plaintext", "pickle"]
+
+
+class AioGraphiteSendException(Exception):
+    pass
+
+
+class AIOGraphite:
+
+	def __init__(self, graphite_server, graphite_port, protocol, loop = None):
+
+		self._graphite_server = graphite_server
+
+		self._graphite_port = graphite_port
+
+		self._graphite_server_address = (graphite_server, graphite_port)
+
+		self._reader, self._writer = None, None
+
+		self.protocol = protocol
+
+		self.loop = loop or asyncio.get_event_loop()
+
+
+	@asyncio.coroutine
+	async def send(self, metric: str, value: int, timestamp = None) -> None:
+		"""
+			@metric: String, valid metric name for Graphite
+			@value: int
+			@timestamp: int
+			Send a single data(metric value timestamp) to graphite
+		"""
+		if not metric:
+			return
+		timestamp = int(timestamp or time.time())
+		# Generate message based on protocol
+		listOfMetricTuples = [self.protocol.data_format(metric, value, timestamp)]
+		message = self.protocol.generate_message(listOfMetricTuples)
+		# Sending Data
+		await self._send_message(message)
+
+
+
+	@asyncio.coroutine
+	async def send_multiple(self, dataset: List[Tuple], timestamp = None) -> None:
```

Comment:
is send_multiple good enough function name or we should rename to send_multiple_messages(). if we change the name then we need to rename send to send_message(). what does everyone think?